ndk-build/src/ndk.rs

```diff
@@ -113,8 +114,16 @@ impl Ndk {
         Ok(Command::new(dunce::canonicalize(path)?))
     }
 
+    /// Returns the highest platform that is supported by the NDK
     pub fn default_platform(&self) -> u32 {
-        self.platforms().iter().max().cloned().unwrap()
+        self.platforms()
+            .iter()
+            .cloned()
+            .rev()
+            // Detect NDK support by existence of the clang compiler, with `android<sdk version>` in the name
+            // TODO: Don't assume some architecture, despite the NDK supporting them all at once
+            .find(|&p| self.clang(Target::Arm64V8a, p).is_ok())
+            .unwrap()
```

Comment:
In hindsight I think this should be moved back to `fn from_env()` to filter everything going into `Self::platforms`, but we can't call `Self::clang()` there yet.

Reply:
Not really that familiar with the code anymore. But looks like `from_env()` would be the place to put it.

Reply:
Yeah, I'm fairly certain that's the way forward. However, we still have the issue with targets. Maybe iterate them all, or just assume every NDK supports `Arm64V8a` for the time being and call it a day? Otherwise we might have to parse some config file to find out what android versions are supported - or base `fn default_platform()` on the selected target triple.

Reply:
how about parsing `build/core/platforms.mk`? I did some grepping and this seems to be the best way of figuring out the supported platforms.

```make
cat build/core/platforms.mk
NDK_MIN_PLATFORM_LEVEL := 16
NDK_MAX_PLATFORM_LEVEL := 30
NDK_PLATFORM_ALIAS_20 := android-19
NDK_PLATFORM_ALIAS_25 := android-24
NDK_PLATFORM_ALIAS_J := android-16
NDK_PLATFORM_ALIAS_J-MR1 := android-17
NDK_PLATFORM_ALIAS_J-MR2 := android-18
NDK_PLATFORM_ALIAS_K := android-19
NDK_PLATFORM_ALIAS_L := android-21
NDK_PLATFORM_ALIAS_L-MR1 := android-22
NDK_PLATFORM_ALIAS_M := android-23
NDK_PLATFORM_ALIAS_N := android-24
NDK_PLATFORM_ALIAS_N-MR1 := android-24
NDK_PLATFORM_ALIAS_O := android-26
NDK_PLATFORM_ALIAS_O-MR1 := android-27
NDK_PLATFORM_ALIAS_P := android-28
NDK_PLATFORM_ALIAS_Q := android-29
NDK_PLATFORM_ALIAS_R := android-30
```

Reply:
Not really fond of parsing makefiles, and we'd have to check how far back this method works if we wish to maintain backwards compatibility of some sort, too.

Reply:
@dvc94ch Done that now, works quite nicely for at least the [currently supported LTS and Rolling NDK releases](https://github.com/android/ndk/wiki#supported-downloads). Will follow up with more test results and a PR documenting supported NDKs shortly, but consider this PR done and ready for review :)